### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by cyrilmottier**
Aka “Shut the fuck up compiler. Stop complaining and compile faster!”

### CODE REVIEW

File: lib/src/main/java/com/snap/ui/recycling/adapter/ObservableViewModelSectionAdapter.kt [hunk 1]
```diff
@@ -106,11 +106,12 @@ open class ObservableViewModelSectionAdapter
             .subscribe(::applyAdapterUpdates)
     }
 
+    @Suppress("UNCHECKED_CAST")
     private fun combineSections(sections: List<ObservableSectionController>): Observable<ItemUpdates> {
         val modelSources = sections.mapIndexed { sectionIndex, section ->
             // Associate each ViewModel with its section
             section.getViewModels()
-                    .map { it as Seekable<AdapterViewModel> }
+                    .map { it }
```
Comment (bjdodson) [inline lib/src/main/java/com/snap/ui/recycling/adapter/ObservableViewModelSectionAdapter.kt:114]: is this even needed?

**Review by bjdodson — APPROVED** [review]

File: lib/src/main/java/com/snap/ui/recycling/adapter/ObservableViewModelSectionAdapter.kt (hunk 1 above)
Comment (cyrilmottier) [reply lib/src/main/java/com/snap/ui/recycling/adapter/ObservableViewModelSectionAdapter.kt:114]: Indeed… did it a bit too naively

Comment (Amir123M) [reply lib/src/main/java/com/snap/ui/recycling/adapter/ObservableViewModelSectionAdapter.kt:114]: Slm

Comment (Amir123M) [reply lib/src/main/java/com/snap/ui/recycling/adapter/ObservableViewModelSectionAdapter.kt:114]: اوك image

Comment (Amir123M) [reply lib/src/main/java/com/snap/ui/recycling/adapter/ObservableViewModelSectionAdapter.kt:114]: Tonight is tired from work

**Pull request closed without merging**